tutorials/2d/custom_drawing_in_2d.rst

```diff
@@ -196,333 +209,850 @@ The same applies to the ``draw_rect`` method with ``filled = false``.
+
+Additional parameters as well as other methods related to text and characters
+can be found on the :ref:`CanvasItem <class_CanvasItem>` class reference.
+
+.. _doc_draw_show_drawing_while_editing_example:
 
-We also increment our angle_from and angle_to values here. However, we must not
-forget to ``wrap()`` the resulting values between 0 and 360°! That is, if the angle
-is 361°, then it is actually 1°. If you don't wrap these values, the script will
-work correctly, but the angle values will grow bigger and bigger over time until
-they reach the maximum integer value Godot can manage (``2^31 - 1``).
-When this happens, Godot may crash or produce unexpected behavior.
+Show the drawing while editing
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 
-Finally, we must not forget to call the ``queue_redraw()`` function, which automatically
-calls ``_draw()``. This way, you can control when you want to refresh the frame.
+While the code so far is able to draw the logo on a running window, it will
+not show up on the ``2D view`` on the editor. In certain cases you would
+also like to show your custom Node2D or control on the editor, to position
+and scale it appropriately, like most other nodes do.
+
+To show the logo directly on the editor (without running it), you can use the
+:ref:`@tool<doc_gdscript_tool_mode>` annotation to request the custom drawing
+of the node to also appear while editing, like this:
 
 .. tabs::
  .. code-tab:: gdscript GDScript
 
-    func _process(delta):
-        angle_from += rotation_angle
-        angle_to += rotation_angle
+    @tool
+    extends Node2D
 
-        # We only wrap angles when both of them are bigger than 360.
-        if angle_from > 360 and angle_to > 360:
-            angle_from = wrapf(angle_from, 0, 360)
-            angle_to = wrapf(angle_to, 0, 360)
-        queue_redraw()
+ .. code-tab:: csharp
+
+    using Godot;
+
+    [Tool]
+    public partial class MyNode2D : Node2D
+
+You will need to save your scene, rebuild your project (for C# only) and reload
+the current scene manually at the menu option ``Scene > Reload Saved Scene``
+to refresh the current node in the ``2D`` view the first time you add or remove
+the ``@tool`` annotation.
+
+Animation
+^^^^^^^^^
+
+If we wanted to make the custom shape change at runtime, we could modify the
+methods called or its arguments at execution time, or apply a transform.
+
+For example, if we want the custom shape we just designed to rotate, we could add
+the following variable and code to the ``_ready`` and ``_process`` methods:
+
+.. tabs::
+ .. code-tab:: gdscript GDScript
+
+    extends Node2D
+
+    @export var rotation_speed : float = 1  # In radians per second.
+
+    func _ready():
+        rotation = 0
+        ...
+
+    func _process(delta: float):
+        rotation -= rotation_speed * delta
 
  .. code-tab:: csharp
 
-    public override void _Process(double delta)
+    [Export]
+    private float _rotationSpeed = 1.0f;  // In radians per second.
```

Comment:
```suggestion
    public float RotationSpeed = 1.0f;  // In radians per second.
```
Not experienced with the C# code but I think this should be public and named like this if exported yes?

Reply:
Oh, sorry, I missunderstood. I missread the suggestion. I also agree it should be public.

Reply:
I see the mistake, I think raul suggested to make it private but it could be an existing mistake. I will apply it and let him comment in case it is a mistake.
![Screenshot_20231220_163629](https://github.com/godotengine/godot-docs/assets/8576860/2518a45a-7889-422a-83b5-5ebb201f707e)

Reply:
I think he misses the `[Export]`, I don't think export should be private, as per the other code 🙂

Reply:
Ah, I suggested `private` because that's the default accessibility when unspecified. I didn't realize this member was exported.

If the member is exported, I agree that it should be public and therefore it should also be a property instead of a field:

```suggestion
    public float RotationSpeed { get; set; } = 1.0f;  // In radians per second.
```

Reply:
Done and tested.